### PR TITLE
MEN-2664: backend-integration tests (GET POST /deployments/[id])

### PR DESCRIPTION
**Pull request by oleorhagen**
Changelog: None

Signed-off-by: Ole Petter <ole.orhagen@northern.tech>

### CODE REVIEW

File: backend-tests/tests/test_deployments.py [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=2)).isoformat("T")) + "Z",
+                        # Batch size is optional in the last stage (ie, it is the remaining devices)
+                    },
+                ],
+            },
+            {
+                "name":
+                "Five phases, first no start time, last start time, no batch size",
```
Comment (mchalski) [inline backend-tests/tests/test_deployments.py]: formatting again - pep or not, this is dumb and confusing, please move to 1 line

File: backend-tests/tests/test_deployments.py [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+                "artifact_name": "deployments-phase-testing",
+                "devices": ["dummyuuid" + str(i) for i in range(100)],
+                "phases": [{
+                    "start_ts": str((datetime.utcnow() +
+                         timedelta(days=1)).isoformat("T")) + "Z",
+                    "batch_size": 100,
+                }],
+            },
+            {
+                "name": "One phase, with start time, and full batch size",
```
Comment (kjaskiewiczz) [inline backend-tests/tests/test_deployments.py]: with start time?

File: backend-tests/tests/test_deployments.py [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+                "artifact_name": "deployments-phase-testing",
+                "devices": ["dummyuuid" + str(i) for i in range(100)],
+                "phases": [{
+                    "start_ts":
+                    str((datetime.utcnow() +
+                         timedelta(days=1)).isoformat("T")) + "Z",
+                }],
+            },
+            {
+                "name": "One phase, with start time",
```
Comment (kjaskiewiczz) [inline backend-tests/tests/test_deployments.py]: one phase?

File: backend-tests/tests/test_deployments.py [hunk 4]
```diff
@@ -0,0 +1,510 @@
...
+                "artifact_name": "deployments-phase-testing",
+                "device_count": 100,
+            }),
+        # One phase:
+        #     - start_time
+        #     + batch_size
+        ({
+            "name": "One phase, with no start time, and full batch size",
```
Comment (kjaskiewiczz) [inline backend-tests/tests/test_deployments.py]: no start time?

File: backend-tests/tests/test_deployments.py (hunk 2 above)
Comment (oleorhagen) [reply backend-tests/tests/test_deployments.py]: Misleading how?

Comment (kjaskiewiczz) [reply backend-tests/tests/test_deployments.py]: the name suggests that there is one phase WITH START TIME and batch size equal to 100, but there is one phase with batch size equal to 100 and WITHOUT START TIME

**Comment by kjaskiewiczz** [comment]
are you testing successful path only?

**Comment by oleorhagen** [comment]
@kjaskiewiczz, yes, we decided that the successful path was enough (me and Marcin)

**Pull request merged**